.github/workflows/build.yml

```diff
@@ -359,15 +359,13 @@ jobs:
 
   release:
     name: Release
-
     if: startsWith(github.ref, 'refs/tags/') || github.event_name == 'workflow_dispatch' && github.event.inputs.publish == 'true'
-
     runs-on: ubuntu-latest
-    environment: Publish
+    environment: pypipublish
     needs: [linux, windows, macos_x86, macos_aarch64, sdist]
     permissions:
-      # IMPORTANT: this permission is mandatory for Trusted Publishing
       id-token: write
+      contents: read
```

Comment:
Added one more permission so we're sure the token can be used, and the contents we're dealing with, read